torch/_dynamo/variables/builtin.py

```diff
@@ -409,6 +409,15 @@ def python_type(self):
     def as_python_constant(self):
         return self.fn
 
+    def as_proxy(self):
+        DTYPE = {
+            int: torch.int64,
```

Comment:
bool and complex?

Also, is this something we already do between the Python and the C++ code in PyTorch eager? If so, it'd be good to have a look at where are we doing that, to see if there's any other logic we may need to implement.

Reply:
Ah, forgot about `bool`. But, `complex` doesn't really work on eager. Should we enable it on dynamo?

Reply:
I think [this](https://github.com/pytorch/pytorch/blob/cb9a4fbbf25e5f15f862d11e206a4028e9898515/torch/csrc/utils/python_arg_parser.h#L730) is the function we are looking for. Adding `bool` seems to do the job.

Reply:
Thank you for digging this up!